template_registry.go

```diff
@@ -89,10 +87,14 @@ func (r *TemplateRegistry) Get(path []string) (t *Template, err error) {
 	if err != nil {
 		return
 	}
-	t, exists := d.content[templateKey]
+	i, exists := d.content[templateKey]
 	if !exists {
 		return nil, fmt.Errorf("template \"%s\" does not exist", strings.Join(path, "/"))
 	}
+	t, ok := i.(*Template)
```

Comment:
I've thought about it, and for me it looks like a more extensible way of fetching typed value from the `interface{}`. For example, if in the future we would like to add `*Link` type alongside existing `*Template` and `*TemplateRegistryDir` types, it would be very handy to have something like that `switch` statement.